File: Practica 3/Pila.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (!isFull()) {
            arreglo[++top] = e;
        } else {
            System.out.println("pila llena");
        }
    }

    //
    public long pop() {
        if (!isEmpty()) {
            return arreglo[top--];
        } else {
            System.out.println("pila vacia");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[top];
        } else {
            System.out.println("pila vacia");
            return -1;
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }
    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        System.out.println("Elemento en la cima: " + pila.peek());
        System.out.println("Sacando elemento: " + pila.pop());
        System.out.println("¿Está vacía?: " + pila.isEmpty());
    }
}

File: Practica 3/Pila.py
class Pila:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.top = -1

    def push(self, e):
        if not self.isFull():
            self.top += 1
            self.arreglo[self.top] = e
        else:
            print("pila llena")

    def pop(self):
        if not self.isEmpty():
            elemento = self.arreglo[self.top]
            self.top -= 1
            return elemento
        else:
            print("pila vacia")
            return -1

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.top]
        else:
            print("pila vacia")
            return -1

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

if __name__ == "__main__":
    pila = Pila(5)
    pila.push(10)
    pila.push(20)
    pila.push(30)
    print("Elemento en la cima:", pila.peek())
    print("Sacando elemento:", pila.pop())
    print("¿Está vacía?:", pila.isEmpty())
